File: src/cs/Library/Elements/CheckBox.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace JJA.Anperi.Lib.Elements
{
    public class CheckBox : Element
    {
        public override string type => "checkbox";

        /// <summary>
        /// This should actually be lowercase but since that's a keyword in c# this has to do.
        /// </summary>
        [JsonProperty("checked")]
        public bool Checked { get; set; }
    }
}

File: src/java/AnperiLib/src/com/jannes_peters/anperi/lib/elements/CheckBox.java
package com.jannes_peters.anperi.lib.elements;

import org.json.simple.JSONObject;

public class CheckBox extends Element {
    public CheckBox(String id, boolean isChecked) {
        super("checkbox", id);
        this.mIsChecked = isChecked;
    }
    private boolean mIsChecked;

    @Override
    protected JSONObject createJSONObject() {
        JSONObject jo = super.createJSONObject();
        jo.put("checked", mIsChecked);
        return jo;
    }
}
